src/private/TitleBar.cpp

```diff
@@ -501,14 +546,32 @@ void TitleBar::onAutoHideClicked()
         return;
     }
 
+    const bool groupedAutoHide = Config::self().flags() & Config::Flag_AutoHideAsTabGroups;
+    const auto currentDw = m_frame->currentDockWidget();
+
     const auto &dockwidgets = m_frame->dockWidgets();
     for (DockWidgetBase *dw : dockwidgets) {
         if (dw->isOverlayed()) {
             // restore
             MainWindowBase *mainWindow = dw->mainWindow();
-            mainWindow->restoreFromSideBar(dw);
+            auto sideBarGroup = groupedAutoHide ? SideBarGroupRegister::groupForDockWidget(dw) : DockWidgetBase::List();
+            if (sideBarGroup.isEmpty()) {
+                mainWindow->restoreFromSideBar(dw);
+            } else {
+                for (auto it = sideBarGroup.rbegin(); it != sideBarGroup.rend(); ++it) {
+                    DockWidgetBase *member = *it;
+                    auto title = member->title();
+                    title.clear();
+                    mainWindow->restoreFromSideBar(member);
+                }
+                dw->setAsCurrentTab();
+            }
+            SideBarGroupRegister::removeDockWidgetFromGroup(dw);
         } else {
-            dw->moveToSideBar();
+            if (groupedAutoHide || dw == currentDw) {
+                dw->moveToSideBar();
+                SideBarGroupRegister::registerDockWidgetsAsGroup(dockwidgets);
```

Comment:
it seems registerDockWidgetsAsGroup() should be done out of the for-loop, otherwise it's registering twice ?

Reply:
> it seems registerDockWidgetsAsGroup() should be done out of the for-loop, otherwise it's registering twice ?

Yes